src/Symfony/Component/Form/LegacyOptionsResolverAdapter.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form;
+
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+/**
+ * This class is only present for backwards compatibility and should not be
+ * used outside the Form component. It will be removed in Symfony 3.0.
+ *
+ * @internal
+ *
+ * @author Christian Flothmann <christian.flothmann@xabbuh.de>
+ */
+class LegacyOptionsResolverAdapter extends OptionsResolver
```

Comment:
this means you must reimplement every public method of OptionsResolver. For example you overwrote `setDefaults` but not `setDefault` which creates huge bugs if one or the other is used.